### PR TITLE
fix compiler warnings

### PR DESCRIPTION
**Pull request by esigo**
Fixes various compiler warnings.

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: api/include/opentelemetry/trace/experimental_semantic_conventions.h [hunk 1]
```diff
@@ -183,7 +183,7 @@ static const struct
 
 inline const char *attr(uint32_t attr)
 {
-  for (int i = 0; i < OTEL_CPP_TRACE_ATTRIBUTES_MAX; i++)
+  for (long unsigned int i = 0; i < OTEL_CPP_TRACE_ATTRIBUTES_MAX; i++)
```
Comment (ThomsonTan) [inline api/include/opentelemetry/trace/experimental_semantic_conventions.h:186]: What is the type `long unsigned int`? Probably `unsigned int`?

Comment (esigo) [reply api/include/opentelemetry/trace/experimental_semantic_conventions.h]: it was the type compiler was showing. I changed to `size_t`.

**Comment by esigo** [comment]
@ThomsonTan I can't reproduce `CI / Bazel valgrind (pull_request)` fail locally. Any idea what's going wrong?

File: exporters/otlp/test/otlp_http_exporter_test.cc [hunk 2]
```diff
@@ -22,6 +22,10 @@
 
 #  if defined(_MSC_VER)
 #    define putenv _putenv
```
Comment (lalitb) [inline exporters/otlp/test/otlp_http_exporter_test.cc:24]: Nit - this can be removed?

File: sdk/test/resource/resource_test.cc [hunk 3]
```diff
@@ -54,6 +55,7 @@ TEST(ResourceTest, create_without_servicename)
       else
         EXPECT_EQ(opentelemetry::nostd::get<std::string>(expected_attributes.find(e.first)->second),
                   opentelemetry::nostd::get<std::string>(e.second));
+    }
```
Comment (lalitb) [inline sdk/test/resource/resource_test.cc:58]: There are a few places in this [test]( https://github.com/open-telemetry/opentelemetry-cpp/blob/4110b56314eb7c3ecbc56b9f6fcb37eb70286858/sdk/test/resource/resource_test.cc#L174 ) using putenv, do you think these can be modified to use setenv? Maybe have a common method here ( https://github.com/open-telemetry/opentelemetry-cpp/blob/main/sdk/include/opentelemetry/sdk/common/env_variables.h ) ?

Comment (esigo) [reply sdk/test/resource/resource_test.cc:58]: those remaining `putenv`s were correct. To be consistent, I replaced all of them with `sentenv`.

**Review by lalitb — APPROVED** [review]

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**